packages/hyperion-autologging/src/ALCustomEvent.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates. All Rights Reserved.
+ */
+
+'use strict';
+
+import type { Channel } from "@hyperion/hook/src/Channel";
+
+import { ALLoggableEvent, ALMetadataEvent, ALOptionalFlowletEvent, ALTimedEvent } from "./ALType";
+
+export type ALCustomEventData =
+  ALMetadataEvent &
+  ALOptionalFlowletEvent &
+  ALTimedEvent &
```

Comment:
I was just thinking for some of these more "core" fields like eventTimestamp and eventIndex etc.. do you think we should control adding them to the event ourselves?  Just so we keep these consistent across all events?

Reply:
Excellent question! I have did separate the module and only added a single type to it now, but my thought was to add a function that will fill out the rest of the fields. I only was concerned about the event index part. I was thinking if people wanted to generate events that may not be loggable, it will increase the event index . 
But now I think we can add an extra functions argument to give control for that as well. I'll add the function and export it externally. 

Note that one can still choose to directly call `channel.emit`, but we can just give them a mechanism for simplicity.